### ISSUE TITLE
Can't follow from Mastodon

### ISSUE THREAD

**Issue by nesges** [bookwyrm-social/bookwyrm#2734]
**Describe the bug**

When I try to follow a bookwyrm user on buecher.pnpde.social from Mastodon the follow request is neither accepted in bookwyrm nor presented to that user. On Mastodon it just remains in "Folgeanfrage zurückziehen" ("withdraw follow request" I guess). Our Celery queues are empty and ai already tryed restarting. Only related logentry I see is:

`web_1             | Unauthorized: /user/D3/inbox`

after trying to follow D3@buecher.pnpde.social

**To Reproduce**
1. Go to Mastodon and search for @D3@buecher.pnpde.social
2. Click on "Folgen" / "Follow"
3. Refresh page
4. Button changes to "Folgeanfrage zurückziehen" / "withdraw follow request"
5. Wait.
6. Request is never accepted

**Expected behavior**
Button should change to "Entfolgen" / "Unfollow" after a short while and remain in that state

**Instance**
https://buecher.pnpde.social

**Additional context**
bookwyrm version is 0.5.4

**Comment by mouse-reeve** [comment]
Are you seeing this for all Mastodon users and all instances, or just this user or users on this instance?

**Comment by nesges** [comment]
I checked with two different bookwyrm users (d3 and moonmoth@buecher.pnpde.social) and from two different mastodon instances (mastodon.pnpde.social and rollenspiel.social)

**Comment by eldang** [comment]
This sounds a lot like the symptoms of the authorized fetch issue which https://github.com/bookwyrm-social/bookwyrm/pull/2613 should fix.  That PR is in the "pre-release" https://github.com/bookwyrm-social/bookwyrm/releases/tag/v0.5.5 and I am wondering if when that is fully released it will fix this problem.

**Comment by nesges** [comment]
I just confirmed a userreport that it's not only mastodon, but also our users cannot be followed from other bookwyrm instances. I could replicate that from on older testing instance on https://wyrm.dnddeutsch.de an from https://wyrms.de

**Comment by nesges** [comment]
I just pulled from the production branch (`git pull`, `bw-dev update`, version is 0.5.5 now). It didn't resolve the issues, but only got rid of the log message i cited above (`web_1 | Unauthorized: /user/D3/inbox`).

**Comment by nesges** [comment]
Seems to be working now! And it boils down to a configuration error I made in nginx. 

We use nginx as reverse proxy and I had the `proxy_set_header Host $host;` commented out. I can't remember exactly why I did that, but I noticed today that bookwyrm returns a 400 bad request error, when I activated that line. Only after adding our domain to `ALLOWED_HOST` (in .env) I was able to use `proxy_set_header Host` and after about an hour of waiting the follow requests are all coming through. Case closed (I guess 🤞)
